pkg/controller/dns/external_dns_test.go

```diff
@@ -52,10 +57,30 @@ var (
 			ZoneIds:       []string{"/subscriptions/subscription/resourceGroups/resourcegroup/providers/Microsoft.Network/privatednszones/test.com"},
 		},
 	}
+	gvr1 = schema.GroupVersionResource{
+		Group:    "group",
+		Version:  "v1",
+		Resource: "resources",
+	}
+	gvk1 = schema.GroupVersionKind{
+		Group:   gvr1.Group,
+		Version: gvr1.Version,
+		Kind:    "Resource",
+	}
 )
 
+func TestMain(m *testing.M) {
+	restConfig, _ = testutils.StartTestingEnv()
+
+	exitCode := m.Run()
+
+	testutils.CleanupTestingEnv()
+	os.Exit(exitCode)
+}
+
 var (
-	self *appsv1.Deployment = nil
+	restConfig *rest.Config
```

Comment:
nit: can we combine this with other var block or at least put it above testmain

Reply:
Done